### ISSUE TITLE
`Qt`

### ISSUE THREAD

**Issue by nelsonic** [dwyl/learn-python#8]
To resolve https://github.com/yorikvanhavre/BIM_Workbench/issues/108 I need to install and run `Qt`:  https://www.qt.io/download-qt-installer-oss

After downloading the `.app` from `qt.io`: 

<img width="512" alt="image" src="https://user-images.githubusercontent.com/194400/231602422-5f997117-73ec-4bd8-be00-92916e5dabab.png">

`macOS` prompts us if we want to open the app:

<img width="535" alt="image" src="https://user-images.githubusercontent.com/194400/231601592-b42b4536-47ac-4622-994e-8bb77e34c220.png">

**Comment by nelsonic** [comment]
Welcome to the `Qt` installer:

<img width="1062" alt="image" src="https://user-images.githubusercontent.com/194400/231602641-1917cd39-eabd-4980-af67-56ee9705ee87.png">

Had to register for an account:

<img width="1062" alt="image" src="https://user-images.githubusercontent.com/194400/231602859-475d1794-f30b-48ec-af52-ef8b71bfe321.png">


Had to verify email address:

<img width="1103" alt="image" src="https://user-images.githubusercontent.com/194400/231603276-13e70c9c-ba99-4f32-a4a5-7d012751e523.png">


Then taken to a page with upsell: 

<img width="1393" alt="image" src="https://user-images.githubusercontent.com/194400/231603220-ba059afd-2b06-44b2-9a30-9fb391663912.png">

**Comment by nelsonic** [comment]
Now I'm allowed to proceed:

<img width="1062" alt="image" src="https://user-images.githubusercontent.com/194400/231603494-69c45f89-b0c6-4cda-aa13-b57a1cd8f14a.png">

Taking some time to doanload:

<img width="1062" alt="image" src="https://user-images.githubusercontent.com/194400/231603585-8fe53f3b-dc49-4cb4-b407-fd3c47ecc513.png">

No thanks to usage stats:

<img width="1062" alt="image" src="https://user-images.githubusercontent.com/194400/231603766-ed181fd6-3a7d-4167-93d1-2f267b01d65d.png">

The default appears to be "Custom Installation":
<img width="1062" alt="image" src="https://user-images.githubusercontent.com/194400/231604341-1455610f-cc7b-490f-89e6-896ee75ecade.png">

Asking the `FreeCAD` Devs what the required install is: https://github.com/yorikvanhavre/BIM_Workbench/issues/108#issuecomment-1506097830

**Comment by nelsonic** [comment]
Tried following the instructions for compiling on `macOs`: https://wiki.freecad.org/Compile_on_MacOS

got the following error:

```sh
==> Installing freecad/freecad/freecad@0.20.1 dependency: freecad/freecad/med-file
==> Patching
==> Applying hdf5-1.12.patch
patching file 'config/cmake_files/medMacros.cmake'
patching file 'src/ci/MEDfileCompatibility.c'
patching file 'src/hdfi/_MEDattributeNumWrByName.c'
patching file 'src/hdfi/_MEDattributeNumWr.c'
patching file 'src/hdfi/_MEDcheckAttributeStringFunc.c'
patching file 'src/hdfi/_MEDchecknSublinkFunc.c'
patching file 'src/hdfi/_MEDdatagroupExist.c'
patching file 'src/hdfi/_MEDdatasetExist.c'
patching file 'src/hdfi/_MEDfileCreate.c'
patching file 'src/hdfi/_MEDfileOpen.c'
patching file 'src/hdfi/_MEDlinkObjs.c'
patching file 'src/hdfi/_MEDmemFileOpen.c'
patching file 'src/hdfi/_MEDparFileCreate.c'
patching file 'src/hdfi/_MEDparFileOpen.c'
patching file 'tools/medimport/2.3.6/_MEDconvertStringDatasets.c'
==> cmake .. -DMEDFILE_BUILD_PYTHON=ON -DMEDFILE_BUILD_TESTS=OFF -DMEDFILE_INSTALL_DOC=OFF -DCMAKE_PREFIX_PATH=/opt/homebrew/opt/hdf5/lib;/opt/homebrew/opt/gcc/lib -DCMAKE_INSTAL
==> make
Last 15 lines from /Users/n/Library/Logs/Homebrew/med-file@4.1.1/02.make:
make[1]: *** [python/CMakeFiles/_medfamily.dir/all] Error 2
[ 96%] Built target medimportengine
[ 96%] Linking C executable mdump4
cd /tmp/med-fileA4.1.1-20230413-31941-1i3rdjh/med-4.1.1_SRC/build/tools/mdump && /opt/homebrew/Cellar/cmake/3.26.3/bin/cmake -E cmake_link_script CMakeFiles/mdump4.dir/link.txt --verbose=1
[ 96%] Linking C executable mdump2
/opt/homebrew/Library/Homebrew/shims/mac/super/clang -O3 -DNDEBUG -arch arm64 -isysroot /Library/Developer/CommandLineTools/SDKs/MacOSX13.sdk -Wl,-search_paths_first -Wl,-headerpad_max_install_names CMakeFiles/mdump4.dir/mdump4.c.o CMakeFiles/mdump4.dir/MED23meshIterators.c.o CMakeFiles/mdump4.dir/MED23fieldIterators.c.o -o mdump4  -Wl,-rpath,/tmp/med-fileA4.1.1-20230413-31941-1i3rdjh/med-4.1.1_SRC/build/src ../../src/libmedC.11.1.1.dylib /opt/homebrew/opt/hdf5/lib/libhdf5.dylib /opt/homebrew/lib/libsz.dylib /Library/Developer/CommandLineTools/SDKs/MacOSX13.sdk/usr/lib/libz.tbd /Library/Developer/CommandLineTools/SDKs/MacOSX13.sdk/usr/lib/libdl.tbd /Library/Developer/CommandLineTools/SDKs/MacOSX13.sdk/usr/lib/libm.tbd
cd /tmp/med-fileA4.1.1-20230413-31941-1i3rdjh/med-4.1.1_SRC/build/tools/mdump && /opt/homebrew/Cellar/cmake/3.26.3/bin/cmake -E cmake_link_script CMakeFiles/mdump2.dir/link.txt --verbose=1
/opt/homebrew/Library/Homebrew/shims/mac/super/clang -O3 -DNDEBUG -arch arm64 -isysroot /Library/Developer/CommandLineTools/SDKs/MacOSX13.sdk -Wl,-search_paths_first -Wl,-headerpad_max_install_names CMakeFiles/mdump2.dir/mdump2.c.o CMakeFiles/mdump2.dir/MED23meshIterators.c.o CMakeFiles/mdump2.dir/MED23fieldIterators.c.o -o mdump2  -Wl,-rpath,/tmp/med-fileA4.1.1-20230413-31941-1i3rdjh/med-4.1.1_SRC/build/src ../../src/libmedC.11.1.1.dylib /opt/homebrew/opt/hdf5/lib/libhdf5.dylib /opt/homebrew/lib/libsz.dylib /Library/Developer/CommandLineTools/SDKs/MacOSX13.sdk/usr/lib/libz.tbd /Library/Developer/CommandLineTools/SDKs/MacOSX13.sdk/usr/lib/libdl.tbd /Library/Developer/CommandLineTools/SDKs/MacOSX13.sdk/usr/lib/libm.tbd
[ 96%] Linking C executable mdump3
cd /tmp/med-fileA4.1.1-20230413-31941-1i3rdjh/med-4.1.1_SRC/build/tools/mdump && /opt/homebrew/Cellar/cmake/3.26.3/bin/cmake -E cmake_link_script CMakeFiles/mdump3.dir/link.txt --verbose=1
/opt/homebrew/Library/Homebrew/shims/mac/super/clang -O3 -DNDEBUG -arch arm64 -isysroot /Library/Developer/CommandLineTools/SDKs/MacOSX13.sdk -Wl,-search_paths_first -Wl,-headerpad_max_install_names CMakeFiles/mdump3.dir/mdump3.c.o CMakeFiles/mdump3.dir/MED23meshIterators.c.o CMakeFiles/mdump3.dir/MED23fieldIterators.c.o -o mdump3  -Wl,-rpath,/tmp/med-fileA4.1.1-20230413-31941-1i3rdjh/med-4.1.1_SRC/build/src ../../src/libmedC.11.1.1.dylib /opt/homebrew/opt/hdf5/lib/libhdf5.dylib /opt/homebrew/lib/libsz.dylib /Library/Developer/CommandLineTools/SDKs/MacOSX13.sdk/usr/lib/libz.tbd /Library/Developer/CommandLineTools/SDKs/MacOSX13.sdk/usr/lib/libdl.tbd /Library/Developer/CommandLineTools/SDKs/MacOSX13.sdk/usr/lib/libm.tbd
[ 96%] Built target mdump4
[ 96%] Built target mdump2
[ 96%] Built target mdump3
make: *** [all] Error 2

If reporting this issue please do so at (not Homebrew/brew or Homebrew/homebrew-core):
  https://github.com/freecad/homebrew-freecad/issues

These open issues may also help:
installing 0.20.1 on Apple M1 (errors installing med-file@4.1.1) https://github.com/FreeCAD/homebrew-freecad/issues/374
```

**Comment by nelsonic** [comment]
<img width="1062" alt="image" src="https://user-images.githubusercontent.com/194400/231608613-5cae0bb1-d8de-4c5a-8e76-438aa0d638be.png">

**Comment by nelsonic** [comment]
Installation will use **`32.38 GB`** of disk space ... 

<img width="1062" alt="image" src="https://user-images.githubusercontent.com/194400/231608660-7a13953a-b00d-4fce-8dcc-c5ac54863b55.png">

This is an _insane_ amount of disk space!! 🤯 
I have the space on my current mac. 🙄 
But I will probably uninstall this when no longer needed because it's absurd! 👎

**Comment by nelsonic** [comment]
Downloading: ⏳ 

<img width="1062" alt="image" src="https://user-images.githubusercontent.com/194400/231609227-4805bf82-36b1-40d0-9772-ce8b093fa761.png">

**Comment by nelsonic** [comment]
**`54 minutes remaining`** indeed ...

<img width="1062" alt="image" src="https://user-images.githubusercontent.com/194400/231610963-129e8157-d960-4869-a3c2-2602ae4583ea.png">

This will take forever to download!! ⏳

**Comment by nelsonic** [comment]
Issue: https://github.com/FreeCAD/homebrew-freecad/issues/374 still unresolved.

**Comment by nelsonic** [comment]
Eventually ... 

![image](https://user-images.githubusercontent.com/194400/231614011-76048b46-dce0-43d2-966f-810de8850601.png)

The following `README.txt` popped up:

![image](https://user-images.githubusercontent.com/194400/231614214-753816e8-667e-448c-bc0d-782dd1c2f54b.png)

Welcome to `Qt Creator`: 
<img width="1372" alt="image" src="https://user-images.githubusercontent.com/194400/231614291-992b5806-8263-4857-aa38-ff37953453e0.png">

I've quit it for now to focus on my work-work.
But will attempt to run the `BIM-workbench` once the work week is done. ⏳

**Comment by nelsonic** [comment]
Based on this comment: https://github.com/yorikvanhavre/BIM_Workbench/issues/108#issuecomment-1506916418 I'm re-installing `Qt Design Studio`: 
![image](https://user-images.githubusercontent.com/194400/232348813-a3d1d100-6532-44db-a258-0c286b25c97f.png)

But obvs it doesn't allow me to overwrite ...
![image](https://user-images.githubusercontent.com/194400/232348866-91609173-ceeb-4fbc-9fdc-f2e57ef098f7.png)

So created a subdirectory: 
![image](https://user-images.githubusercontent.com/194400/232348906-ad394266-6bf1-46c1-a395-f59b6bb7090e.png)

**Comment by nelsonic** [comment]
But how do I *open* `Qt` after it's installed? 🤷‍♂️
